set.mm

```diff
@@ -662596,6 +663406,1223 @@ closed under the multiplication ( ' X ' ) of a finite number of
       ALVMEFHIACFBUTGJVDKVEVF $.
   $}
 
+  ${
+    $d k m x y $.
+    $( If (/) is a complex number, then it converges to itself.  (see ~ 0ncn
+       and its comment ; see also the comment in ~ climlimsupcex ) (Contributed
+       by Glauco Siliprandi, 2-Jan-2022.) $)
+    0cnv $p |- ( (/) e. CC -> (/) ~~> (/) ) $=
```

Comment:
> See discussion group thread.

Which thread? Can you please post a link?

set.mm has

|- -. (/) e. CC

but it is a construction-dependent property, in fact new usage is discouraged.

Here is Norman's comment in ~0ncn

"Note: do not use this after the real number axioms are developed, since it is a construction-dependent property."

Thus, actually we don't have the theorem

|- -. (/) e. CC

On the contrary,

|- ( (/) e. CC -> (/) ~~> (/) )

would be proven with any arbitrary construction of the complex numbers (it is construction independent). That's why I put it in.

Reply:
This one: https://groups.google.com/g/metamath/c/BXJq5o5i8a8/m/WowODKouCgAJ

After looking at df-clim, I think I can see what you're doing (I haven't looked at your proof yet, I'll do it when it's merged).  In the expression (/) ~~> (/), the (/) on the left is the empty function, the (/) on the right is a complex number, and you're probably using the fact that a function evaluates outside its domain to the empty set (which is due to the particular encoding used in set.mm), so this is abusing the encoding used in set.mm. It's not meant to be a criticism at all, but I think we agree there is no mathematical content in this theorem ?  Of course feel free to leave it as is in your mathbox.  If you think I'm missing something, I'll look at the proof once on the website.

Reply:
> This one: https://groups.google.com/g/metamath/c/BXJq5o5i8a8/m/WowODKouCgAJ
> 
> In the expression (/) ~~> (/), the (/) on the left is the empty function, the (/) on the right is a complex number

Exactly. I was investigating the behavior of ~~> when applied to the empty function.

Thinking about it, I guess I could complete it with another theorem like

|- ( -. (/) e. CC -> -.  (/) e. dom ~~> )

that is, "assuming that the empty set is not a complex number, the empty function doesn't converge".

(but I've not proven it yet, so I'm not sure about it).

I absolutely agree, there's no mathematical content in this theorem, just a couple of assertions to clarify the behavior of our particular definition of convergence when applied to the empty set.
Of course, our definition of convergence agrees with what we find in textbooks for the "standard" case, i.e. sequences, but with metamath we often have to deal with edge cases that textbooks overlook, so it might be useful to write it once and keep if for future reference: if, in the future, someone will have to deal with an edge case and will search for "(/) ~~>" , she will see that it depends on an "undecidable" antecedent and look no further.

Reply:
w.r.t. the thread, I asked about

-. +oo e. CC be

because, while for -. (/) e. CC it is clear to me Norman's intent to state it as something that we CANNOT rely upon (no textbook will ever "take a position" about the empty set being a complex number or not), it was not so clear to me if math literature had an "official" position about chosing +oo and -oo distinct not only from the reals, but from the complex numbers too.

Reply:
Since we don't want the empty function to converge to anything, your theorem shows that we'd better require that -. (/) e. CC, as is rightly the case (or that the value of a function out of its domain be not the empty set). Similarly, we'd better have +oo not in any other set that could come up, to avoid clashes and "accidental" or "junk" theorems (for instance: "plus infinity belongs to the algebraic closure of RR"!!). See for instance https://groups.google.com/g/metamath/c/GQmelsKtaWQ